presto-main/src/main/java/com/facebook/presto/operator/aggregation/FloatRegressionAggregation.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.RegressionState;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.type.SqlType;
+
+import static com.facebook.presto.operator.aggregation.AggregationUtils.getRegressionIntercept;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.getRegressionSlope;
+import static com.facebook.presto.spi.type.FloatType.FLOAT;
+import static java.lang.Float.floatToRawIntBits;
+import static java.lang.Float.intBitsToFloat;
+
+@AggregationFunction("")
+public class FloatRegressionAggregation
+{
+    private FloatRegressionAggregation() {}
+
+    @InputFunction
+    public static void input(RegressionState state, @SqlType(StandardTypes.FLOAT) long dependentValue, @SqlType(StandardTypes.FLOAT) long independentValue)
+    {
+        DoubleRegressionAggregation.input(state, intBitsToFloat((int) dependentValue), intBitsToFloat((int) independentValue));
+    }
+
+    @CombineFunction
+    public static void combine(RegressionState state, RegressionState otherState)
+    {
+        DoubleRegressionAggregation.combine(state, otherState);
+    }
+
+    @AggregationFunction("regr_slope")
+    @OutputFunction(StandardTypes.FLOAT)
+    public static void regrSlope(RegressionState state, BlockBuilder out)
+    {
+        double result = getRegressionSlope(state);
+        if (Double.isFinite(result)) {
```

Comment:
Yes, but then you cast the value to a float which could change a finite value into infinity. It seems like you should cast first and then check whether it's finite